planning/autoware_obstacle_cruise_planner/src/node.cpp

```diff
@@ -1454,38 +1506,17 @@ std::optional<StopObstacle> ObstacleCruisePlannerNode::createStopObstacleForPred
   const auto & p = behavior_determination_param_;
   const auto & object_id = obstacle.uuid.substr(0, 4);
 
-  // NOTE: consider all target obstacles when driving backward
   if (!isStopObstacle(obstacle.classification.label)) {
     return std::nullopt;
   }
-
   const double max_lat_margin_for_stop =
     (obstacle.classification.label == ObjectClassification::UNKNOWN)
       ? p.max_lat_margin_for_stop_against_unknown
       : p.max_lat_margin_for_stop;
 
+  // Obstacle not inside of trajectory check predicted path of the obstacle
```

Comment:
May I ask you to refactor `createStopObstacleForPredictedObject()` ?
I would like to reduce duplicate code even if we have to modify the existing code.

Reply:
Thank you!  I'll response by the next Wednesday. sorry for the late response.